database/database_functions.py

```diff
@@ -1,64 +1,96 @@
+        with regards to person name and person type and the room they belong to into a dictionary.
+        '''
+        # calling the initialize function to use the specified the database
+        self.initialize(filename)
+        allocations = self.SESSIONS.query(Persons).all()
+        allocations_dictionary = {}
+        for allocation in allocations:
+            allocations_dictionary.update({allocation.person_name:[allocation.office_name,
+                                                                   allocation.livingspace_name]})
+        return allocations_dictionary
 
-def add_person(person_name, person_type):
-    ''' Adding a person to database'''
-    newperson = Persons(p_name=person_name, p_type=person_type)
-    SESSIONS.add(newperson)
-    SESSIONS.commit()
+    def add_person(self, filename, person_name, person_type, office_name, livingspace_name):
+        '''
+        The function is used to add the person to the dictionary
+        '''
+        # calling the initialize function to use the specified the database
+        self.initialize(filename)
+        newperson = Persons(person_name=person_name, person_type=person_type,
+                            office_name=office_name, livingspace_name=livingspace_name)
+        self.SESSIONS.add(newperson)
+        self.SESSIONS.commit()
 
-def add_room(room_name, room_type, room_capacity):
-    ''' Adding a room to database'''
-    newuser = Rooms(r_name=room_name, r_type=room_type, r_capacity=room_capacity)
-    SESSIONS.add(newuser)
-    SESSIONS.commit()
+    def add_room(self, filename, room_name, room_type):
+        '''
+        The function is used to add the rooms to the dictionary
+        '''
+        # calling the initialize function to use the specified the database
+        self.initialize(filename)
+        newuser = Rooms(room_name=room_name, room_type=room_type)
+        self.SESSIONS.add(newuser)
+        self.SESSIONS.commit()
 
-def allocate_room(room_name, person_type):
-    ''' Adding a person to database'''
-    newallocation = sessions.query(Rooms).filter_by(r_name = room_name).one()
-    newallocation.r_occupied +=1
-    SESSIONS.add(newpallocation)
-    SESSIONS.commit()
 
-def add_allocation(person_name, office_name, livingspace_name):
-    ''' Updating allocations table '''
-    new_allocation = Allocations(p_name=person_name, o_name=office_name, ls_name=livingspace_name)
-    SESSIONS.add(new_allocation)
-    SESSIONS.commit()
-
-def edit_allocation(p_name, office_name, livingspaces_name):
-    relocation = sessions.query(Rooms).filter_by(p_name = p_name).one()
-    relocation.o_name = office_name
-    relocation.ls_name = livingspaces_name
+    def update_person(self, filename, person_name, office_name, livingspaces_name):
+        '''
+        The function is used to add the person to the dictionary
+        '''
+        # calling the initialize function to use the specified the database
```

Comment:
This comment is not really necessary, because there is a docstring explaining what the function does